### ISSUE TITLE
Support for platforms, lifecycles, and toolchains YAML files

### ISSUE THREAD

**Issue by samrocketman** [samrocketman/jervis#142]
JSON has been great... but YAML is more flexible and user-friendly than JSON for admins to add support.

JSON will always be supported for compatibility.  However, it would be nice to have YAML as an option.

Example JSON files.

* platforms.json
* lifecycles-ubuntu1604-stable.json
* toolchains-ubuntu1604-stable.json

And the same purpose files but in YAML.

* platforms.yaml
* lifecycles-ubuntu1604-stable.yaml
* toolchains-ubuntu1604-stable.yaml

This is becomes more important as support for documentation generation gets implemented in the future.  Refer to issue #139.

**Comment by samrocketman** [comment]
With upcoming 2.0 release YAML will be the primary format going forward.  JSON support for backwards compatibility.

**Comment by samrocketman** [comment]
https://github.com/samrocketman/jervis/blob/main/vars/hasGlobalResource.groovy was added for this purpose.  e.g. detect YAML or fall back to JSON.

**Comment by samrocketman** [comment]
https://github.com/samrocketman/jervis/commit/8da9edc97b3521103cf495b7303bb47da52c62fb `hasGlobalResource` enables this.

**Comment by samrocketman** [comment]
Convert platforms JSON to YAML

```
import net.gleske.jervis.tools.YamlOperator

Map platforms = YamlOperator.loadYamlFrom(new File('resources/platforms.json'))

new File('resources/platforms.yaml').withWriter { w ->
    w << YamlOperator.writeObjToYaml(platforms)
}
```

**Comment by samrocketman** [comment]
Closed by 8c9320476760ca59195ab888bdba2090e48497c6
